woocommerce/woocommerce-android
# Issue/5555 line chart

Closes: #5555

Merge this PR first: #5626 and then merge this into Trunk

This is ready for review, but I'd prefer not to merge it until this task: #5616 is completed as well

### Description
Convert home tab bar chart to line chart. Main UI changes to be applied:

- Convert the chart view from bar chart to line chart
- Disable circles for data points in the line chart (enabled by default, where each data point has a circle drawn)
- Update the color and width of the data line and highlighted line
- Create a new `MarkerImage` "dot" for the circle marker shown when highlighting a data point (the pink circle in the screenshots above)
- Add a gradient to the line chart. The gradient is only enabled for positive values. Line chart values under 0 x axis won't display any gradient. 
- Updated [MPAndroidChart](https://github.com/PhilJay/MPAndroidChart) to the latest version and upadate ValueFormatters and usages of deprecated methods

### Testing instructions
Just open the Home Screen tab and check that the line chart is displayed properly without any UI glitches. 

### Images/gif

https://user-images.githubusercontent.com/2663464/148798765-2cbdf0b9-ec64-42ad-9349-be4967f08016.mp4
